### PR TITLE
Add invoice list view and order detail

### PR DESCRIPTION
**Pull request by adguernier**
- [x] Create custom Invoice list view
- [x] Add order details in list view
image

### CODE REVIEW

**Review by fzaninotto — APPROVED** [review]
What's left for RFR? Please provide a checklist in the PR description

File: src/ui/invoice/InvoiceShow.tsx [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+      <Card sx={{ minWidth: 275 }}>
+        <CardContent>
+          <Stack
+            direction="row"
+            spacing={2}
+            justifyContent="space-between"
+            alignItems="center"
+          >
+            <Typography variant="h6">Posters Galore</Typography>
+            <Typography variant="h6">Invoice {record.id}</Typography>
```
Comment (fzaninotto) [inline src/ui/invoice/InvoiceShow.tsx]: Why don't you use a `<TextField>` to render te id, instead of relying on `useRecordContext`?

Comment (adguernier) [reply src/ui/invoice/InvoiceShow.tsx]: Ho yes ! it was from an older commit

File: src/ui/order/Basket.tsx [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+  if (isLoading) return <Loading />;
+
+  if (!products) return null;
+
+  const representation = products.map((product) =>
+    Object.assign(
+      product,
+      order.basket.find((b) => b.product_id === product.id)
+    )
+  );
```
Comment (fzaninotto) [inline src/ui/order/Basket.tsx]: I don't understand this...

**Review by fzaninotto — CHANGES_REQUESTED** [review]

File: src/ui/order/Basket.tsx (hunk 2 above)
Comment (adguernier) [reply src/ui/order/Basket.tsx]: This is for getting data of products and order; together

This result in that : 
```json
[
  {
    "id": 108,
    "category_id": 10,
    "reference": "Eye Rest",
    "width": 26.15,
    "height": 30.87,
    "price": 41.68,
    "thumbnail": "https://marmelab.com/posters/tech-9.jpeg",
    "image": "https://marmelab.com/posters/tech-9.jpeg",
    "description": "Temporibus consequatur sunt praesentium qui. Amet doloremque illo. Minima nobis suscipit voluptas.",
    "stock": 0,
    "sales": 18,
    "product_id": 108,
    "quantity": 1
  }
]
```
Maybe there is an easier way ?

File: .eslintrc.json [hunk 3]
```diff
@@ -19,6 +18,7 @@
   "plugins": ["@typescript-eslint"],
   "rules": {
     "@typescript-eslint/ban-ts-comment": "warn",
-    "react/react-in-jsx-scope": "off"
+    "react/react-in-jsx-scope": "off",
+    "react/jsx-key": "off"
```
Comment (djhi) [inline .eslintrc.json]: Why, react do need keys

File: src/ui/order/Basket.tsx (hunk 2 above)
Comment (djhi) [reply src/ui/order/Basket.tsx]: In any case, assign to an empty object at least to avoid mutating `product`

**Review by djhi — CHANGES_REQUESTED** [review]

File: .eslintrc.json (hunk 3 above)
Comment (adguernier) [reply .eslintrc.json]: Because of this 
image

Comment (djhi) [reply .eslintrc.json]: Then no, disable the rule inline but not globally otherwise you might miss some actual mistakes

**Review by djhi — APPROVED** [review]

**Pull request merged**